tests/test_resilient_requests.py

```diff
@@ -0,0 +1,17 @@
+# -*- coding: utf-8 -*-
+
+import requests
+
+from hepdata_cli.resilient_requests import TimeoutHTTPAdapter, retry_strategy
+from hepdata_cli.api import SITE_URL
+
+
+def test_timeout():
+    retry_strategy.total = 2
+    adapter = TimeoutHTTPAdapter(max_retries=retry_strategy, timeout=0.0001)
+    with requests.Session() as session:
+        session.mount("https://", adapter)
+        try:
+            session.get(SITE_URL + '/ping')
+        except requests.exceptions.ConnectTimeout:
+            pass
```

Comment:
[`pytest.raises`](https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions) could be helpful here to check that the exception is actually raised.

Reply:
Done!